### DISCUSSION TITLE
Streaming / displaying 360 videos

### DISCUSSION THREAD

**Discussion by JeromeCHA** [sceneview/sceneview#366] [Q&A]
Hi,

Currently, by using `environment` and `HDRLoader.loadEnvironment`, it is possible to display hdr 360 files.
However, as I mentionened in the title, I would like to know if there is a good approach to display / showing a 360 videos (or 360 live webrtc streaming videos)?

I saw that there is a [`Stream`](https://sceneview.github.io/api/sceneview-android/sceneview/io.github.sceneview.texture/-video-texture/-builder/stream.html) in [`VideoTexture`](https://sceneview.github.io/api/sceneview-android/sceneview/io.github.sceneview.texture/-video-texture/-builder/index.html)'s builder, but is it the good way to do it?

Thank you for your help 🙇

**Comment by JeromeCHA** [comment]
Looks like nobody check the discussion tabs 😅

**Comment by ThomasGorisse** [comment]
Hi @JeromeCHA.

The `Skybox` is based on a cubemap environement `Texture` so yes, you might be able to stream what you want to it.
Have a look at:
- [Skybox.Builder](https://github.com/google/filament/blob/95ecc3d1e411432339bfe9b82baa656818bc623e/android/filament-android/src/main/java/com/google/android/filament/Skybox.java#L74-L94) 
- [VideoTexture](https://sceneview.github.io/api/sceneview-android/sceneview/io.github.sceneview.texture/-video-texture/-builder/index.html)
- [VideoMaterial](https://github.com/SceneView/sceneview-android/blob/c3b8c5a0fca0dbb43c971a91fe808456f51cd65e/sceneview/src/main/java/io/github/sceneview/material/VideoMaterial.kt) will guide you on how to get a `Texture` from a `MediaPlayer`
- [LightEstimator](https://github.com/SceneView/sceneview-android/blob/aa8eda5408a7dae2b99d019f01f1cf08b129a8b4/arsceneview/src/main/java/io/github/sceneview/ar/light/LightEstimator.kt#L185-L254) can show you how to get an `Environment` from your cubemap. If minSDK > 26, GPU harware buffer might be more efficient.

**Reply by JeromeCHA** [reply]
Hi @ThomasGorisse Thank you for your reply! 😃 

Let me try it and come back later ! 🙇

**Comment by ThomasGorisse** [comment]
And if you want to have fun. and have one Camera per cubemap face, you can do what `ARScene` does with a single face and be welcome to the cars upper camera view world.

**Reply by JeromeCHA** [reply]
@ThomasGorisse Thank you for your advice. However, I don't get it. 
Can you explain how to get a cubemap from a VideoTexture?

I tried with `IBLPrefilter`, but it doesn't work because VideoTexture is not a 2D texture. It is quite confusing even looking at the code.

https://github.com/SceneView/sceneview-android/blob/ad5237503650d15bf780353aa8de159b7a08c295/sceneview_2_0_0/src/main/java/io/github/sceneview/utils/IBLPrefilter.kt#L32-L48

**Reply by JeromeCHA** [reply]
Since the answer of the original question has been answered, I will close it andI will open a new discussion :)
